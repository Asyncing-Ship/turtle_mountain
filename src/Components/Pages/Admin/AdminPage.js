import React, { Component } from 'react';

class AdminPage extends Component {
  render() {
    return (
      <>
        This is the admin page!
      </>
    )
  }
}

export default AdminPage;