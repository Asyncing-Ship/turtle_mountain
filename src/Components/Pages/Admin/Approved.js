import React from "react";
import {
  Box,
  Button,
  useDisclosure,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalBody,
  ModalFooter,
  AlertDialog,
  AlertDialogOverlay,
  AlertDialogContent,
  AlertDialogHeader,
  AlertDialogBody,
  AlertDialogFooter,
  Tag,
  TagIcon,
  TagLabel,
} from "@chakra-ui/core";
import { connect } from "react-redux";

const Approved = (props) => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  const [isOpenAlert, setIsOpen] = React.useState();
  const onCloseAlert = () => setIsOpen(false);
  const cancelRef = React.useRef();

  return (
    <>
      <Box
        textAlign="right"
        rounded="lg"
        borderWidth="1px"
        p={2}
      >
        <Box
          as="span"
          textAlign="left"
          verticalAlign="middle"
          ml={3}
        >
          {
            props.user.is_admin ?
              <Tag
                rounded="full"
                size="sm"
                variantColor="purple"
                mr={2}
              >
                <TagIcon icon="star" size="10px" />
                <TagLabel>{props.user.first_name} {props.user.last_name}</TagLabel>
              </Tag>
              :
              <Tag
                rounded="full"
                size="sm"
                variantColor="yellow"
                mr={2}
              >
                <TagLabel>{props.user.first_name} {props.user.last_name}</TagLabel>
              </Tag>
          }
        </Box>
        <Box as="span" verticalAlign="middle" ml={3}>
          {!props.user.is_admin ? (
            <Button
              size="sm"
              variantColor="purple"
              rightIcon="star"
              onClick={onOpen}
            >
              Promote
            </Button>
          ) : (
              <Button
                size="sm"
                variantColor="yellow"
                rightIcon="minus"
              >
                Demote
              </Button>
            )
          }
        </Box>
        <Box
          as="span"
          verticalAlign="middle"
          ml={3}
          alignItems="right"
          justifyContent="right"
          textAlign="right"
        >
          <Button
            size="sm"
            variantColor="red"
            rightIcon="close"
            onClick={() => setIsOpen(true)}
          // onClick={() => }
          >
            Remove
          </Button>
        </Box>

        {/* Delete User Alert /////////////////////////////////////////////////////////////////////////////////////////////*/}

        <>
          <AlertDialog
            isOpen={isOpenAlert}
            leastDestructiveRef={cancelRef}
            onClose={onClose}
          >
            <AlertDialogOverlay />
            <AlertDialogContent rounded="lg">
              <AlertDialogHeader fontSize="lg" fontWeight="bold">
                Delete Member
              </AlertDialogHeader>

              <AlertDialogBody>
                You are about to remove member{" "}
                {
                  props.user.is_admin ?
                    <Tag
                      rounded="full"
                      size="sm"
                      variantColor="purple"
                      mr={2}
                    >
                      <TagIcon icon="star" size="10px" />
                      <TagLabel>{props.user.first_name} {props.user.last_name}</TagLabel>
                    </Tag>
                    :
                    <Tag
                      rounded="full"
                      size="sm"
                      variantColor="yellow"
                      mr={2}
                    >
                      <TagLabel>{props.user.first_name} {props.user.last_name}</TagLabel>
                    </Tag>
                }
                from Turtle Mountain Connect. This cannot be undone.
              </AlertDialogBody>

              <AlertDialogFooter>
                <Button
                  size="sm"
                  ref={cancelRef}
                  onClick={onCloseAlert}
                >
                  Cancel
                </Button>
                <Button
                  size="sm"
                  variant="ghost"
                  variantColor="red"
                  onClick={() => {
                    onCloseAlert();
                    props.dispatch({
                      type: "DELETE_USER",
                      payload: props.user.id,
                    });
                  }}
                  ml={3}
                >
                  Delete Member
                </Button>
              </AlertDialogFooter>
            </AlertDialogContent>
          </AlertDialog>
        </>

        {/* Promote User Modal ////////////////////////////////////////////////////////////////////////////////////////////*/}

        <Modal isOpen={isOpen} onClose={onClose}>
          <ModalOverlay />
          <ModalContent rounded="lg">
            <ModalHeader>Modify Role</ModalHeader>
            <ModalBody>
              <p>
                You are about to grant member{" "}
                <Tag
                  rounded="full"
                  size="sm"
                  variantColor="yellow"
                  mr={2}
                >
                  <TagLabel>{props.user.first_name} {props.user.last_name}</TagLabel>
                </Tag>
                all admin privileges.
              </p>
            </ModalBody>

            <ModalFooter>
              <Button size="sm" mr={3} onClick={onClose}>
                Cancel
              </Button>
              <Button
                size="sm"
                variant="ghost"
                variantColor="purple"
                onClick={() => {
                  onClose();
                  props.dispatch({
                    type: "PROMOTE_USER",
                    payload: props.user.id,
                  });
                }}
              >
                Promote to Admin
              </Button>
            </ModalFooter>
          </ModalContent>
        </Modal>
      </Box>
    </>
  );
};

export default connect()(Approved);
