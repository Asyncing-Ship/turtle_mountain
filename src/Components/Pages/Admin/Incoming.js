import React from "react";

import {
  Box,
  Button,
  ButtonGroup,
  Tag,
  TagLabel,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalBody,
  ModalFooter,
  useDisclosure
} from "@chakra-ui/core";
import { connect } from "react-redux";
const Incoming = (props) => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <Box
      textAlign="right"
      rounded="lg"
      borderWidth="1px"
      p={2}
    >
      <Box
        as="span"
        textAlign="left"
        verticalAlign="middle"
        ml={3}
      >
        <Tag
          rounded="full"
          size="sm"
          variantColor="gray"
          mr={2}
        >
          <TagLabel>{props.user.first_name} {props.user.last_name}</TagLabel>
        </Tag>
      </Box>
      <ButtonGroup ml={3}>
        <Button
          size="sm"
          variantColor="green"
          rightIcon="check"
          onClick={onOpen}
        >
          Approve
        </Button>
        <Button
          size="sm"
          variantColor="red"
          rightIcon="close"
          onClick={() => {
            props.dispatch({
              type: "DELETE_USER",
              payload: props.user.id,
            });
          }}
        >
          Reject
        </Button>
      </ButtonGroup>

      {/* Approve User Modal //////////////////////////////////////////////////////////////////////////////////////// */}

      <Modal isOpen={isOpen} onClose={onClose}>
        <ModalOverlay />
        <ModalContent rounded="lg">
          <ModalHeader>New User Detected</ModalHeader>
          <ModalBody>
            <p>
              You are about to grant{" "}
              <Tag
                rounded="full"
                size="sm"
                variantColor="gray"
                mr={2}
              >
                <TagLabel>{props.user.first_name} {props.user.last_name}</TagLabel>
              </Tag>
                regular access as a member.
              </p>
          </ModalBody>

          <ModalFooter>
            <Button size="sm" mr={3} onClick={onClose}>
              Cancel
              </Button>
            <Button
              size="sm"
              variant="ghost"
              variantColor="green"
              onClick={() => {
                onClose();
                props.dispatch({
                  type: "APPROVE_USER",
                  payload: props.user.id,
                });
              }}
            >
              Approve New Member
              </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </Box>
  );
};

export default connect()(Incoming);
