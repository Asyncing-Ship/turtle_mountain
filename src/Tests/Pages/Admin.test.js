import React from "react";
import { shallow } from "enzyme";
import AdminPage from "../../Components/Pages/Admin/AdminPage";
import Approved from "../../Components/Pages/Admin/Approved";
import Incoming from "../../Components/Pages/Admin/Incoming";
import toJson from "enzyme-to-json";
import { createStore } from "redux";
import rootReducer from "../../Redux/Reducers/reducers";
const store = createStore(rootReducer);
it("renders without crashing", () => {
  shallow(<AdminPage />);
});
it("renders without crashing", () => {
  const mockColor = "David";
  const wrapper = shallow(<AdminPage color={mockColor} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders without crashing", () => {
  shallow(<Approved />);
});
it("renders without crashing", () => {
  const mockColor = "David";
  const wrapper = shallow(<Approved color={mockColor} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders without crashing", () => {
  shallow(<Incoming />);
});
it("renders without crashing", () => {
  const mockColor = "David";
  const wrapper = shallow(<Incoming color={mockColor} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
