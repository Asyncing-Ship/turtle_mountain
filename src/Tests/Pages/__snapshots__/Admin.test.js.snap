// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  This is the admin page!
</Fragment>
`;

exports[`renders without crashing 2`] = `
<Fragment>
  These people are approved:
</Fragment>
`;

exports[`renders without crashing 3`] = `
<Fragment>
  These are incoming user requests:
</Fragment>
`;
